frame/authorship/src/lib.rs

```diff
@@ -321,7 +342,9 @@ impl<T: Config> Pallet<T> {
 		let now = <frame_system::Pallet<T>>::block_number();
 
 		let mut uncles = <Uncles<T>>::get();
-		uncles.push(UncleEntryItem::InclusionHeight(now));
+		uncles
+			.try_push(UncleEntryItem::InclusionHeight(now))
+			.expect("the list of uncles accepted per generation is bounded, and the number of generations is bounded, so pushing a new element will always succeed");
```

Comment:
Feels like this should just return an error instead of panicking.

Reply:
We have a `defensive_proof` option that will allow you to specify a reason why it should never return an error, and it's defined in `frame_support::traits::Defensive`.